lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1399,69 +1413,61 @@ export class UserAgentApplication {
+    hash = this.getHash(hash);
+    return Utils.deserialize(hash);
+  }
+
+  /**
+   * Creates a stateInfo object from the URL fragment and returns it.
    * @param {string} hash  -  Hash passed from redirect page
    * @returns {TokenResponse} an object created from the redirect response from AAD comprising of the keys - parameters, requestType, stateMatch, stateResponse and valid.
    * @ignore
    * @hidden
    */
-  // TODO: Change to return AuthResponse object, rename this, it is not request but response
-  protected getRequestInfo(hash: string): TokenResponse {
-    hash = this.getHash(hash);
-    const parameters = Utils.deserialize(hash);
-    const tokenResponse = new TokenResponse();
-
-    // If there is a hash with parameters in the redirect response
-    if (parameters) {
-      tokenResponse.parameters = parameters;
-
-      // Check for error or tokens
-      if (parameters.hasOwnProperty(Constants.errorDescription) ||
-        parameters.hasOwnProperty(Constants.error) ||
-        parameters.hasOwnProperty(Constants.accessToken) ||
-        parameters.hasOwnProperty(Constants.idToken)
-      ) {
-        tokenResponse.valid = true;
-
-        // Identify the callback based on the state
-        let stateResponse: string;
-        if (parameters.hasOwnProperty("state")) {
-          stateResponse = parameters.state;
-        } else {
-          return tokenResponse;
-        }
-
-        tokenResponse.stateResponse = stateResponse;
-        // async calls can fire iframe and login request at the same time if developer does not use the API as expected
-        // incoming callback needs to be looked up to find the request type
-
-        // loginRedirect
-        if (stateResponse === this.cacheStorage.getItem(Constants.stateLogin, this.inCookie) || stateResponse === this.silentAuthenticationState) { // loginRedirect
-          tokenResponse.requestType = Constants.login;
-          tokenResponse.stateMatch = true;
-          return tokenResponse;
-        }
-        // acquireTokenRedirect
-        else if (stateResponse === this.cacheStorage.getItem(Constants.stateAcquireToken, this.inCookie)) { //acquireTokenRedirect
-          tokenResponse.requestType = Constants.renewToken;
-          tokenResponse.stateMatch = true;
-          return tokenResponse;
-        }
-
-        // external api requests may have many renewtoken requests for different resource
-        if (!tokenResponse.stateMatch) {
-          tokenResponse.requestType = window.requestType;
-          const statesInParentContext = window.renewStates;
-          for (let i = 0; i < statesInParentContext.length; i++) {
-            if (statesInParentContext[i] === tokenResponse.stateResponse) {
-              tokenResponse.stateMatch = true;
-              break;
-            }
-          }
+  protected getResponseState(hash: string): StateResponse {
+    const parameters = this.stringifyHash(hash);
+    let stateResponse: StateResponse;
```

Comment:
as mentioned stateResponse alludes to a different meaning.  "responseState"??

Reply:
Changing to ResponseStateInfo